tqdm/_tqdm.py

```diff
@@ -716,12 +797,13 @@ def __iter__(self):
                 # Note: does not call self.update(1) for speed optimisation.
                 n += 1
                 # check the counter first (avoid calls to time())
-                if n - last_print_n >= miniters:
+                if n - last_print_n >= self.miniters:
+                    miniters = self.miniters  # watch monitoring thread changes
```

Comment:
This is an optimization, we store instance variables as local function variables, this speeds up the calculations a great deal because there's no dot reference lookup. But here we need to access self.miniters since it can be modified by the monitoring thread.